parlai/core/metrics.py

```diff
@@ -115,39 +127,157 @@ class AverageMetric(Metric):
     Class that keeps a running average of some metric.
     """
 
-    def __init__(self, numer: TScalar, denom: TScalar):
-        self._numer = self.as_float(numer)
-        self._denom = self.as_int(denom)
+    __slots__ = ('_numer', '_denom')
 
-    def __add__(self, other: 'AverageMetric') -> 'AverageMetric':
+    def __init__(self, numer: TScalar = 0, denom: TScalar = 1):
```

Comment:
Hmm - why these values for defaults? I could see a situation in which I forget one of these values and then get an unexpected number as a result without warning

Reply:
Numer could go without a default. Denom's default i think is very reasonable.

Reply:
Yeah, up to you, but personally I like the idea of leaving at least one arg without a default for each metric to avoid the inevitable silent error from forgetting to pass in any args at all. For SumMetric this wouldn't change the value at all so maybe it's not necessary, but for AverageMetric it would